### DISCUSSION TITLE
Job logs load very slowly and cause browser to freeze with a very large group output

### DISCUSSION THREAD

**Discussion by joxl** [community/community#8848] [Actions]
My team recently implemented [grouping Workflow commands][grouping_log_lines] in one of our Workflow actions (code test runner action) which always generates a large amount of logging output.  The action has a few distinct sections, and splitting them into collapsible groups of output improves job navigation and output readability.

Adding the output grouping has caused a very poor user experience while viewing the action if it fails.  Navigating to the failed action initially displays the grouped sections that did not fail, and then Workflows takes a very long time to display the remaining Action output on the screen (and subsequently scroll to the bottom).

Note that if the action fails we intentionally do not emit the `::endgroup::` Workflows command.  This is to prevent the "offending output" (i.e. the output that includes details about the failure) from collapsing.

This issue is related to the viewport (and ultimately screen) size when viewing a failed action.  If your viewport is sufficiently small that the "failed output group" is not in view when navigating to the failed job, the failed output immediately starts loading (without freezing the browser) with more output loading as-needed when you scroll.  If your viewport is sufficiently large that the _entire action_ is in view when navigating to the failed job, the output prior to the "failed output group" is immediately displayed, but then the page freezes for a long time (many seconds) until suddenly the entire failed output is displayed all at once and the page automatically scrolls to the bottom.

## Steps to reproduce:

1. Open a new browser window and resize until the viewport has a relatively small height (1200x580 was used for the example link provided in the step 4).
2. Ensure the window zoom level is 100%.
3. Navigate to https://github.com and login.
4. Navigate to [this failing Workflows job][failing_job].

    **OBSERVE (EXPECTED)**:

    The failed `Run tests` action is in view, but only the first line of output from the action is visible (because the window is small). The output loads quickly and you can immediately begin scrolling down through the (many lines of) action output text.

5. Resize the same browser window so that the `[run.sh] INFO: Django test suite: python` text (in yellow) is in view when the page is scrolled to the top (viewport size 1200x860 was used for this example).
6. Refresh the page.

    **OBSERVE**:

    The `Run tests` action loads, initially only showing the (collapsed) group titled `Django test suite: python`, no further action output is shown beyond this point, and the other actions (after the `Run tests` action) are visible in view.  The browser window (tab) is now unresponsive and may emit a warning asking the user if they would like to stop the unresponsive script.  Example:

    ![image](https://user-images.githubusercontent.com/1731363/146219715-c9e32a36-911d-436f-a72f-92454d8ee2f4.png)

    **EXPECTED**:

    The `Run tests` action loads quickly and allows the user to immediately begin scrolling the output.

Above steps reproduced on macOS 10.15 (Catalina) using latest version (at time of post) of Mozilla Firefox (95.0, 64-bit).

[grouping_log_lines]: https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#grouping-log-lines
[failing_job]: https://github.com/dimagi/commcare-hq/runs/4489770227?check_suite_focus=true

**Comment by ethomson** [comment]
Thanks for the feedback and the detailed reproduction steps.  This is reasonably fast for me on macOS Safari, so this may be a bit of a browser oddity that we may need to consider.  I'll pass this along to the team.  Thanks again.

**Reply by joxl** [reply]
I didn't try Safari on macOS, but I did try both Firefox and Chrome on macOS and Ubuntu Linux. The issue affects all browsers/OS's I've tried.
